### PR TITLE
include more complete remark transform example

### PR DESCRIPTION
**Pull request by ChristianMurphy**
addresses https://github.com/fkling/astexplorer/pull/550#discussion_r527588560

this adds an example transform that increases the depth of headings by 1, `# header` to `## header`, `## header` to `### header` etc.

### CODE REVIEW

**Comment by wooorm** [comment]
Can we somehow add a package in ASTExplorer?
It would be a bit nicer if folks can use the common package `unist-util-visit` to walk the tree

**Comment by wooorm** [comment]
semicolon use is inconsistent btw

**Comment by ChristianMurphy** [comment]
> Can we somehow add a package in ASTExplorer?

In ASTExplorer yes.
Into examples, not that I have found (yet).
in examples both `import` and `require` are limited/unavailible.

**Comment by fkling** [comment]
I’m not on my computer right now, but it’s possible to make additional packages available to transforms, it just needs to be preconfigured. There is another transform that does that I think.
I can provide more info when I’m at my computer.

**Comment by fkling** [comment]
The second argument to `compileModule` allows you to specify which variables should be available in the transform. You can either do what the postcss transform does and [provide a fake `require` function](https://github.com/fkling/astexplorer/blob/860b77e178e2cff9fbb041a1b38527524dd3cd0a/website/src/parsers/css/transformers/postcss/index.js#L22-L32), or you can the tools available directly, [like babel-plugin-macros](https://github.com/fkling/astexplorer/blob/860b77e178e2cff9fbb041a1b38527524dd3cd0a/website/src/parsers/js/transformers/babel-plugin-macros/index.js#L24-L27).

**Comment by ChristianMurphy** [comment]
Thanks for the tip @fkling! :bow: 
I've included three common utils in 152ae00 to simplify transform testing.

File: website/src/parsers/markdown/transformers/remark/index.js [hunk 1]
```diff
@@ -1,5 +1,20 @@
 import compileModule from '../../../utils/compileModule';
 import pkg from 'remark/package.json';
+import unistUtilIs from 'unist-util-is';
+import unistUtilVisit from 'unist-util-visit';
+import unistUtilVisitParents from 'unist-util-visit-parents';
```
Comment (fkling) [inline website/src/parsers/markdown/transformers/remark/index.js:5]: I'd prefer if those are also loaded dynamically, but if they are not too large then it's OK for now. You could create the `sandboxRequire` function in the `loadTransformer` method to make it available inside `transform`.

File: website/src/parsers/markdown/transformers/remark/codeExample.txt [hunk 2]
```diff
@@ -1,5 +1,8 @@
+const visit = require("unist-util-visit");
```
Comment (fkling) [inline website/src/parsers/markdown/transformers/remark/codeExample.txt:1]: I think we should include a comment here that explains which packages can be required.

File: website/src/parsers/markdown/transformers/remark/index.js (hunk 1 above)
Comment (ChristianMurphy) [reply website/src/parsers/markdown/transformers/remark/index.js]: addressed in 0e12f3c

File: website/src/parsers/markdown/transformers/remark/codeExample.txt (hunk 2 above)
Comment (ChristianMurphy) [reply website/src/parsers/markdown/transformers/remark/codeExample.txt:2]: addressed in 0e12f3c

**Comment by fkling** [comment]
Awesome. It's live now :slightly_smiling_face:

**Pull request merged**